server/server/src/static_files.rs

```diff
@@ -1,28 +1,48 @@
 use std::io::ErrorKind;
+use std::io::Write;
 use std::sync::Mutex;
 
 use actix_files as fs;
+use actix_multipart::Multipart;
 use actix_web::error::InternalError;
 use actix_web::http::header::{ContentDisposition, DispositionParam, DispositionType};
+use actix_web::http::StatusCode;
 use actix_web::web::Data;
 use actix_web::{get, guard, web, Error, HttpRequest, HttpResponse};
-use reqwest::StatusCode;
-use serde::Deserialize;
+use futures_util::TryStreamExt;
+use repository::sync_file_reference_row::SyncFileReferenceRowRepository;
+use serde::{Deserialize, Serialize};
+
+use repository::sync_file_reference_row::SyncFileReferenceRow;
+
 use service::plugin::plugin_files::{PluginFileService, PluginInfo};
 use service::plugin::validation::ValidatedPluginBucket;
+use service::service_provider::ServiceProvider;
 use service::settings::Settings;
-use service::static_files::StaticFileService;
+use service::static_files::{StaticFileCategory, StaticFileService};
 
 // this function could be located in different module
 pub fn config_static_files(cfg: &mut web::ServiceConfig) {
     cfg.service(web::resource("/files").guard(guard::Get()).to(files));
     cfg.service(plugins);
+    cfg.service(
+        web::resource("/sync_files/{table_name}/{record_id}")
```

Comment:
Is `sync_files` a good path name here?

Reply:
it's growing on me - my initial reaction: didn't love it.
Partly because 'sync' is more commonly a verb, when the intention is an adjective i.e. 'synchronised files' and I read it as 'synchronise files'
naming is tricky - I would have gone just 'files' which is already taken. 
so.. yeah? keep it as 'sync_files' 🤷

Reply:
We could actually use `files` I think? But Iiked having a separation between sync files and the report downloads, as well as keep the code simple. I guess we could probably get away with changing the other one to something else? like `/tmp_files/`?

Reply:
I like the separation too.. if anything, the report `/files` should change, since that it really `/reports`